### PR TITLE
script to verify hutch name

### LINKED ISSUE #106: Create verify_hutch or similar function/script
We should have a script that verifies valid hutch names. Probably lowercase default and uppercase by option? This can then be used in other engineering_tools scripts instead of the assortment of checks we currently have built in. Using this would also be an improvement over the "unknown_hutch" checks in imgr and other scripts.

_Originally posted by @ZLLentz in https://github.com/pcdshub/engineering_tools/pull/101#discussion_r806127397_

### PR DESCRIPTION
**Pull request by c-tsoi**
## Description

Wrote a helper script to verify hutches, argument passed in is a string exits with 0 if successful, 1 if not successful.

## Motivation and Context

Fixes #106.

## How Has This Been Tested?

I tested the script with false hutch names and edge cases including first and last entries.

## Where Has This Been Documented?

Usage.

### CODE REVIEW

File: scripts/verify_hutch [hunk 1]
```diff
@@ -0,0 +1,46 @@
...
+if [ $# -lt 1 ]; then
+   echo 'need arguments: input ioc or pv name' >&2
+   usage
+   exit 1
+fi 
+
+if [[ ($1 == "--help") || ($1 == "-h") ]]; then
+   usage
+   exit 0
+fi
```
Comment (ZryletTC) [inline scripts/verify_hutch:28]: I would suggest moving this  to right below the usage

File: scripts/verify_hutch [hunk 2]
```diff
@@ -0,0 +1,46 @@
...
+usage: ${BASH_SOURCE[0]} <ioc>|<pv> [option]
+
+Script that returns confirmation if the passed hutch is a known/existing hutch
+
+EOF
+}
+
+#hutches=['tmo','txi','rix','xpp','xcs','mfx','cxi','mec', 'ued', 'det', 'lfe','kfe','tst', 'las', 'hpl']
+
+foundHutch=False
```
Comment (ZryletTC) [inline scripts/verify_hutch:16]: For bash, I typically recommend just 0 and 1 instead of False and True

**Review by ZryletTC — COMMENTED** [review]
Looking good. I'd just say clean up your comments and debug messages, then there's a bit of unnecessary code that can be taken out as well. Otherwise, good to go.

File: scripts/verify_hutch [hunk 3]
```diff
@@ -0,0 +1,46 @@
...
+HUTCH=${1,,}
```
Comment (ZryletTC) [inline scripts/verify_hutch:17]: I recommend lowercase variable name for script-specific variables and just to match that its a lower case hutch name
```suggestion
hutch=${1,,}
```

File: scripts/verify_hutch [hunk 4]
```diff
@@ -0,0 +1,39 @@
+#!/bin/bash
+
+# function that takes in a string and verifies its a known hutch
+
+usage(){
+cat << EOF
+usage: ${BASH_SOURCE[0]} <hutch> [option]
```
Comment (ZryletTC) [inline scripts/verify_hutch:7]: Other than -h and --help, this script doesn't take any options
```suggestion
usage: ${BASH_SOURCE[0]} <hutch>
```

File: scripts/verify_hutch [hunk 5]
```diff
@@ -0,0 +1,39 @@
+#!/bin/bash
+
+# function that takes in a string and verifies its a known hutch
+
```
Comment (ZryletTC) [inline scripts/verify_hutch:4]: The comment up here is a bit redundant with the usage directly following it
```suggestion
```

File: scripts/verify_hutch [hunk 6]
```diff
@@ -0,0 +1,39 @@
...
+usage(){
+cat << EOF
+usage: ${BASH_SOURCE[0]} <hutch> [option]
+
+Script that returns confirmation if the passed hutch is a known/existing hutch, success indicated by exit 0, failure by exit 1. 
+
+EOF
+}
+
+foundHutch=1
```
Comment (ZryletTC) [inline scripts/verify_hutch:14]: False is typically 0, and True is 1

File: scripts/verify_hutch [hunk 7]
```diff
@@ -0,0 +1,39 @@
...
+if [ $# -lt 1 ]; then
+   echo 'need arguments: input hutch name' >&2
+   usage
+   exit 1
+fi 
+
+if [[ ($1 == "--help") || ($1 == "-h") ]]; then
+   usage
+   exit 0
+fi
```
Comment (ZryletTC) [inline scripts/verify_hutch:25]: Just changing spacing and order to organize a bit better, argument checking should go before the variable assignments.
```suggestion
if [ $# -lt 1 ]; then
   echo 'need arguments: input hutch name' >&2
   usage
   exit 1
fi 
if [[ ($1 == "--help") || ($1 == "-h") ]]; then
   usage
   exit 0
fi

foundHutch=1
hutch=${1,,}

```

File: scripts/verify_hutch [hunk 8]
```diff
@@ -0,0 +1,39 @@
...
+    elif [[ $hutch == $i ]]; then
+        foundHutch=0
+	exit 0
+    else 
+        foundHutch=1
+    fi
+done
+if [[ $foundHutch == 1 ]]; then
+    exit 1
+fi
```
Comment (ZryletTC) [inline scripts/verify_hutch:38]: This section can be condensed a good amount. Try to find which lines aren't necessary and take them out :)

**Review by ZryletTC — COMMENTED** [review]
Looking better, keep improving 👍

**Review by ZLLentz — APPROVED** [review]
The script looks solid to me! I'll give out my green checkmark.

One thing to consider: should we also add an entry to the table in the repository's README.md file?

**Comment by ZryletTC** [comment]
Yes, @c-tsoi please add the usage for this file to the README.md (it's sorted alphabetically by script name), then I'll approve as well :)

File: README.md [hunk 9]
```diff
@@ -565,6 +565,13 @@ Takes a run with dark images for use in pedestals, and posts to the elog.
     </td>
 </tr> 
 
+<tr>
+    <td>verify_hutch</td>
+    <td>
+usage: Verifies that the passed argument is a known hutch, exit 0 for true or exit for false.
```
Comment (ZryletTC) [inline README.md:571]: Copy and pasting from your usage within the script (and changing `${BASH_SOURCE[0]}` to `verify_hutch`
```suggestion
usage: verify_hutch <hutch>
 Script that returns confirmation if the passed hutch is a known/existing hutch, success indicated by exit 0, failure by exit 1. 
```

**Review by ZryletTC — COMMENTED** [review]
Changing usage to same as from inside the script.
Also before we merge, could you please change the script name to `verify-hutch` as I believe that has been decided as the best naming convention.

**Review by ZryletTC — APPROVED** [review]
LGTM! 👍

**Pull request merged**